### PR TITLE
No longer start navbar dropdown with a divider

### LINKED ISSUE #1319: separator included before first entry in Articles navbar dropdown
Let me know if a reprex would be useful. As an example:

https://github.com/rstudio/renv/blob/da66b72a77a1708053f5cc659db6b0396ac99e0c/_pkgdown.yml#L66-L86

This generates the navbar:

Screen Shot 2020-05-04 at 3 22 44 PM

IMHO the first separator (displayed before `Introduction`) should be removed.

### PR DESCRIPTION
**Pull request by maelle**
Fix #1319 but in BS4

image

### CODE REVIEW

**Comment by hadley** [comment]
Can we add a test for this?

File: R/navbar.R [hunk 1]
```diff
@@ -170,33 +170,6 @@ menu_spacer <- function() {
   menu_text("---------")
 }
 
-
-# Testing helpers ---------------------------------------------------------
```
Comment (maelle) [inline R/navbar.R:174]: move this to the end as I had included some lines that are not testing helpers

File: R/navbar.R [hunk 2]
```diff
@@ -246,7 +219,13 @@ bs4_navbar_links_tags <- function(links, depth = 0L) {
     }
 
     if (!is.null(x$text) && grepl("^\\s*-{3,}\\s*$", x$text)) {
+
+      if (index == 1) {
```
Comment (maelle) [inline R/navbar.R:223]: this is the important change.

File: tests/testthat/_snaps/navbar.md [hunk 3]
```diff
@@ -1,3 +1,112 @@
...
+
+    reference:
+      text: Reference
+      href: reference/index.html
+    intro:
+      text: Get started
+      href: test.html
+    
+
+# can control articles navbar through articles meta
```
Comment (maelle) [inline tests/testthat/_snaps/navbar.md:38]: converted from verify output

File: tests/testthat/test-navbar.R [hunk 4]
```diff
@@ -90,3 +84,42 @@ test_that("data_navbar()can remove elements", {
...
+  expect_snapshot(cat(render_navbar_links(x, bs_version = 3)))
+  expect_snapshot(cat(render_navbar_links(x, bs_version = 4)))
+})
+
+test_that("render_navbar_links BS4 no divider before first element", {
+  x <- list(
+    articles = list(
+      text = "Articles",
+      menu = list(
+        list(text = "---------"),
```
Comment (maelle) [inline tests/testthat/test-navbar.R:114]: first divider gets ignored

File: R/navbar.R (hunk 2 above)
Comment (hadley) [reply R/navbar.R:223]: I'd use an `else` block just to make it clear that there are two equal-weight paths.

**Review by hadley — APPROVED** [review]

**Pull request merged**